sysdig/resource_sysdig_secure_organization_test.go

```diff
@@ -0,0 +1,132 @@
+//go:build tf_acc_sysdig_secure || tf_acc_sysdig_common
+
+package sysdig_test
+
+import (
+	"bytes"
+	b64 "encoding/base64"
+	"encoding/json"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"os"
+	"regexp"
+	"testing"
+
+	"github.com/draios/terraform-provider-sysdig/sysdig"
+)
+
+func TestAccSecureOrganization(t *testing.T) {
+	// XXX: TF acceptance tests for secure org onboarding need an actual existing gcp project
+	// along with an actual service_principal_key to scrape all folders and projects under the org.
+	// Without it POST /organizations call will fail with 500 error.
+	// Skipping the test based on this error when it occurs.
+	rText := func() string { return acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum) }
+	accID := rText()
+	organizationApiUrl := fmt.Sprintf(`%s/api/cloudauth/v1/organizations`, os.Getenv("SYSDIG_SECURE_URL"))
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck: func() {
+			if v := os.Getenv("SYSDIG_SECURE_API_TOKEN"); v == "" {
+				t.Fatal("SYSDIG_SECURE_API_TOKEN must be set for acceptance tests")
+			}
+		},
+		ProviderFactories: map[string]func() (*schema.Provider, error){
+			"sysdig": func() (*schema.Provider, error) {
+				return sysdig.Provider(), nil
+			},
+		},
+		ErrorCheck: func(err error) error {
+			// if regex matches with the expected error, do t.Skip
+			re := regexp.MustCompile(fmt.Sprintf(`POST %s giving up after 5 attempt(s)`, organizationApiUrl))
```

Comment:
are there any other error paths which need to be handled here?

Reply:
Not that I'm aware of. Besides we would want the test to catch other errors, such as 400 Bad Request etc.
Anything else you think we should handle?

Reply:
Resolving this. If required to add handling of specific errors, it can be a follow-up PR.